src/js/components/TabForm.js

```diff
@@ -49,18 +54,30 @@ class TabForm extends React.Component {
     this.setState({currentTab});
   }
 
+  handleFormError() {
+    this.internalStorage_update({isFormValidated: false});
+  }
```

Comment:
the reason for this is kind of complicated, and it's the best solution i can think of so far.

`TabForm` is kind of a special form because it actually renders many `Form`s. When you call a submit on `TabForm`, it then iterates through the `Forms` and calls submit on each of them. if one of those `Forms` error out, we need to set some flag that there was an error, so that when we're done iterating through the `Form`s, we can either decide to call `onError` or `onSubmit`

we don't call `setState` because we don't need to re-render and `setState` is async in some mysterious ways. doing this guarantees it to be synchronous

i'm not sure i got my point 100% across but i don't know of a better solution